rero_ils/modules/acquisition/rollover.py

```diff
@@ -0,0 +1,664 @@
+            data['acq_account']['$ref'] = get_ref_for_pid('acac', p_acc_pid)
+            # Update specific order line fields
+            data['quantity'] = data.unreceived_quantity
+            del data['total_amount']
+
+            # Create the new order line.
+            #   If create failed :: raise an error.
+            #   If success :: fill the mapping table AND the stack of new obj.
+            try:
+                new_line = AcqOrderLine.create(
+                    data, dbcommit=True, reindex=True, delete_pid=True)
+                self.stack.append(new_line)
+                self.mapping_table['order_lines'][line.pid] = new_line.pid
+                old_label = truncate(str(line), 55).ljust(57)
+                new_label = truncate(str(new_line), 55)
+                log.info(f"\t* migrate {old_label} --> {new_label}")
+            except Exception as e:
+                raise RolloverError(f'Order line creation failed :: {str(e)}')\
+                    from e
+
+    def _update_budgets(self, orig_state=False, dest_state=False):
+        """Update rollover budgets to activate/deactivate them.
+
+        :param orig_state (boolean): the new state for original budget.
+        :param dest_state (boolean): the new state for destination budget.
+        """
+        self.logger.info("\tUpdating budget resources...")
+        orig_data = deepcopy(self.original_budget)
+        orig_data['is_active'] = orig_state
+        self.original_budget.update(orig_data, dbcommit=True, reindex=True)
+        state_str = 'activated' if orig_state else 'deactivated'
+        self.logger.info(f"\t  * Original budget is now {state_str}")
+
+        dest_data = deepcopy(self.destination_budget)
+        dest_data['is_active'] = dest_state
+        self.destination_budget.update(dest_data, dbcommit=True, reindex=True)
+        state_str = 'activated' if dest_state else 'deactivated'
+        self.logger.info(f"\t  * Destination budget is now {state_str}")
+
+    # PRIVATE METHODS =========================================================
+    #  These methods are used during the rollover process. They shouldn't be
+    #  use outside this class
+
+    def _abort_rollover(self, message=None):
+        """Aborting the rollover process.
+
+        This will delete all acquisition resources created on the destination
+        `Budget` resource in the reverse order of their creation.
+
+        :param message: the message to log.
+        """
+        if message:
+            self.logger.warning(message)
+        self.logger.warning('Aborting rollover process !')
+        self._update_budgets(True, False)
+        if not self.stack:
+            return
+        self.logger.info('Purging created resources...')
+        for obj in reversed(self.stack):
+            obj.delete(force=True, dbcommit=True, delindex=True)
+            self.logger.info(f"\t* object {str(obj)} deleted")
+
+    def _confirm(self, question, default="yes"):
```

Comment:
I would use a `--yes-i-know`, for ex. https://github.com/inveniosoftware/invenio-indexer/blob/dde365bf12b7c175ec834f305ab52a8b644a2d10/invenio_indexer/cli.py#L84.